fe/src/main/java/org/apache/doris/catalog/Catalog.java

```diff
@@ -5035,6 +5074,35 @@ public void replayRenameColumn(TableInfo tableInfo) throws DdlException {
         throw new DdlException("not implmented");
     }
 
+    public void modifyTableDynamicPartition(Database db, OlapTable table, Map<String, String> properties) throws DdlException {
+        Map<String, String> analyzedDynamicPartition = DynamicPartitionUtil.analyzeDynamicPartition(db, table, properties);
+        TableProperty tableProperty = table.getTableProperty();
+        if (tableProperty != null) {
```

Comment:
I think even if tableProperty is null, we can create a new table property to set dynamic partition properties, just like adding dynamic partition properties to a table at runtime.